File: languages/java/src/main/java/com/bloock/sdk/entity/identity/DatetimeAttribute.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

/**
 * Represents an attribute with a datetime value.
 */
public class DatetimeAttribute extends Attribute<LocalDateTime> {
  /**
   * Creates a new DatetimeAttribute instance with the provided key and value.
   * 
   * @param id
   * @param value
   */
  public DatetimeAttribute(String id, LocalDateTime value) {
    super(id, value);
  }

  public static DatetimeAttribute fromProto(BloockIdentityEntities.DateTimeAttribute res) {
    LocalDateTime parsedDate = LocalDateTime.parse(res.getValue());
    return new DatetimeAttribute(res.getId(), parsedDate);
  }

  public BloockIdentityEntities.DateTimeAttribute toProto() {
    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
    String formattedDate = this.value.format(formatter);

    return BloockIdentityEntities.DateTimeAttribute.newBuilder()
        .setId(this.id)
        .setValue(formattedDate)
        .build();
  }
}

File: languages/go/entity/identityV2/datetime_attribute.go
package identityV2

import (
	"time"

	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
)

// DatetimeAttribute represents an attribute with a datetime value.
type DatetimeAttribute struct {
	Attribute
}

// NewDatetimeAttribute creates a new DatetimeAttribute instance with the provided key and value.
func NewDatetimeAttribute(key string, value time.Time) DatetimeAttribute {
	a := Attribute{
		Id:    key,
		Value: value.Format(time.RFC3339),
	}
	return DatetimeAttribute{
		Attribute: a,
	}
}

func NewDatetimeAttributeFromProto(s *proto.DateTimeAttributeV2) DatetimeAttribute {
	if s == nil {
		return DatetimeAttribute{}
	}
	a := Attribute{
		Id:    s.GetId(),
		Value: s.GetValue(),
	}
	return DatetimeAttribute{
		Attribute: a,
	}
}

func (s DatetimeAttribute) ToProto() *proto.DateTimeAttributeV2 {
	return &proto.DateTimeAttributeV2{
		Id:    s.Attribute.Id,
		Value: s.Attribute.Value.(string),
	}
}
